Lib/test/test_ntpath.py

```diff
@@ -637,6 +638,36 @@ def test_realpath_cwd(self):
         with os_helper.change_cwd(test_dir_short):
             self.assertPathEqual(test_file_long, ntpath.realpath("file.txt"))
 
+    @unittest.skipUnless(HAVE_GETFINALPATHNAME, 'need _getfinalpathname')
+    def test_realpath_permission(self):
+        # Test whether python can resolve the real filename of a
+        # shortened file name even if it does not have permission to access it.
+        ABSTFN = ntpath.abspath(os_helper.TESTFN)
```

Comment:
These needs to be a real path for the comparison to be reliable.

```suggestion
        ABSTFN = ntpath.realpath(os_helper.TESTFN)
```

Reply:
Right. Fixed.